spec.md

```diff
@@ -0,0 +1,100 @@
+# bimg File Format Specification
+bimg (abbreviation of "blit image") is a versatile image format for the ComputerCraft Minecraft mod.
+This specification aims to provide a standardized image format that can be easily implemented and used throughout the entire ComputerCraft ecosystem.
+
+# File Structure
+The examples shown throughout this specification are based on a sample image file, serialized using `textutils.serialize`.
+
+An image file can be separated into two sections: metadata and frame data:
+![Format overview](images/file-structure.png?raw=true)
+
+Metadata contains information about the image itself, while the remaining of the file contains one or more frames, representing the images.
+
+For more information on metadata, see [Metadata](#metadata). For more information on the image data and frames, see [Image Data](#image-data)
+.
+# Metadata
+bimg stores the following metadata along the image data in the file.
+
+| Field | Type | Description |
+|:--|:-:|:--|
+| `version` | `string` | Format version of the file. Used to differentiate two bimg file formats. Composed of three numbers separated by dots, defining the major, minor, and revision number respectively. |
```

Comment:
We should specify the value of the version field in the document to avoid confusion.